compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -133,13 +133,16 @@ class ExtractDependencies extends Phase {
     if (depFile != null) {
       // Cannot ignore inheritance relationship coming from the same source (see sbt/zinc#417)
       def allowLocal = dep.context == DependencyByInheritance || dep.context == LocalDependencyByInheritance
-      val depClassFile =
-        if depFile.isClass then depFile
-        else depFile.resolveSibling(dep.to.binaryClassName + ".class")
-      if (depClassFile != null) {
-        // Dependency is external -- source is undefined
-        processExternalDependency(depClassFile, dep.to.binaryClassName)
-      } else if (allowLocal || depFile != sourceFile.file) {
+      if depFile.isTastyExtension then
+        val depClassFile = ctx.getSiblingClassfile(depFile)
+        if depClassFile != NoAbstractFile then
+          // did not find associated class file, e.g. for a TASTy-only classpath.
+          // The file that Zinc recieves with binaryDependency is used to lookup any either any
+          // generated non-local classes or produced xsbti.API associated with the file.
```

Comment:
Does incremental compilation works at all with a tasty-only classpath? What happens if we just always pass to sbt the name of a .class file even if it doesn't exist?

Reply:
> What happens if we just always pass to sbt the name of a .class file even if it doesn't exist?

I guess that would lead to overcompilation:
https://github.com/lampepfl/dotty/blob/a73316af791502b82525471816fe9153c36f845a/compiler/src/dotty/tools/dotc/core/SymDenotations.scala#L536-L540
But what happens if we pass the .tasty file instead?